### ISSUE TITLE
How do I integrate NotGrid with Clique?

### ISSUE THREAD

**Issue by karai17** [gnwl/NotGrid#46]
I am trying to use NotGrid but I keep getting an issue where my spells are only casting on me, not the unit I am clicking on. Because CLique only works with spells from the spellbook, or custom scripts, I assumed that I should write custom scripts for my spells so that SmartHealer is correctly triggered. I put the following in a custom script:

```
SmartHealer:CastHeal("Greater Heal")
```

which is the equivalent of:

```
/heal Greater Heal
```

but it seems that these custom scripts do not like slash commands, so I am directly calling the function instead. The spell rank selection seems to be working, but it only casts spells on me rather than whom I click.

**Comment by karai17** [comment]
Replaced SmartHealer with LazySpell and I am able to integrate with Clique correctly, but I am still having the issue where it is only casting on me and not on the target. It is correctly picking the right rank as if I were to cast on the target, but it ends up casting on me.

**Comment by karai17** [comment]
Looks like I have to disable "Auto Self Cast" in the interface options to get this to work correctly, which is a bit of a pain. I wonder if there is any way around that?

**Comment by gnwl** [comment]
I don't think there really is, and it's out of scope for me anyway as I prefer to delegate those things to other addons. The reason why it, and likely all mouseover/clickcast addons, does this is because the way it works is it will save your desired target, clear whatever your currrent target is, queue up a spell as if it were on the cursor ready to be clicked on a unit to cast it on, then cast it on the desired target. Then it will restore your previous target.

If you have auto self cast on when it goes to queue up a spell to cast, wow auto self casts onto yourself.

Basically, the example here: https://wowwiki-archive.fandom.com/wiki/API_SpellTargetUnit

There is no "cast desired spell on desired unitID" api function, so this is the best these addons can do.
